src/CarbonAware.DataSources/CarbonAware.DataSources.Json/test/JsonDataSourceTests.cs

```diff
@@ -74,6 +74,24 @@ public void GetCurrentCarbonIntensityForecastAsync_NotImplemented()
 
     }
 
+    [Test]
+    public async Task GetCarbonIntensityAsync_ReturnsData_UsingSamplingWindow()
```

Comment:
Consider naming this test case for the bug condition it's resolving:

```suggestion
    public async Task GetCarbonIntensityAsync_ReturnsSingleDataPoint_WhenStartParamExactlyMatchesDataStart()
```